src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceProperties.cs

```diff
@@ -135,9 +87,84 @@ public GatewayIPAddressInformationCollection GetGatewayAddresses()
                     string gatewayIPHex = parser.MoveAndExtractNext();
                     long addressValue = Convert.ToInt64(gatewayIPHex, 16);
                     IPAddress address = new IPAddress(addressValue);
-                    collection.InternalAdd(new LinuxGatewayIPAddressInformation(address));
+                    collection.InternalAdd(new SimpleGatewayIPAddressInformation(address));
+                }
+            }
+
+            return collection;
+        }
+
+        private IPAddressCollection GetDhcpServerAddresses()
+        {
+            // Parse the /var/lib/dhcp/dhclient.leases file, if it exists.
+            // If any errors occur, like the file not existing or being
+            // improperly formatted, just bail and return an empty collection.
+            InternalIPAddressCollection collection = new InternalIPAddressCollection();
+            try
+            {
+                string fileContents = File.ReadAllText(NetworkFiles.DHClientLeasesFile);
+                int leaseIndex = -1;
+                int secondBrace = -1;
+                while ((leaseIndex = fileContents.IndexOf("lease", leaseIndex + 1)) != -1)
+                {
+                    int firstBrace = fileContents.IndexOf("{", leaseIndex);
+                    secondBrace = fileContents.IndexOf("}", leaseIndex);
+                    int blockLength = secondBrace - firstBrace;
+
+                    int interfaceIndex = fileContents.IndexOf("interface", firstBrace, blockLength);
+                    int afterName = fileContents.IndexOf(';', interfaceIndex);
+                    int beforeName = fileContents.LastIndexOf(' ', afterName);
+                    string interfaceName = fileContents.Substring(beforeName + 2, afterName - beforeName - 3);
+                    if (interfaceName != _linuxNetworkInterface.Name)
+                    {
+                        continue;
+                    }
+
+                    int indexOfDhcp = fileContents.IndexOf("dhcp-server-identifier", firstBrace, blockLength);
+                    int afterAddress = fileContents.IndexOf(";", indexOfDhcp);
+                    int beforeAddress = fileContents.LastIndexOf(' ', afterAddress);
+                    string dhcpAddressString = fileContents.Substring(beforeAddress + 1, afterAddress - beforeAddress - 1);
+                    IPAddress dhcpAddress;
+                    if (IPAddress.TryParse(dhcpAddressString, out dhcpAddress))
+                    {
+                        collection.InternalAdd(dhcpAddress);
+                    }
                 }
             }
+            catch
+            {
+                // If any parsing or file reading exception occurs, just ignore it and return the collection.
```

Comment:
(But I think an IOException would cover either of those cases, i.e. file doesn't exit, or file can't be parsed)